usd/src/sdf2usd_TEST.cc

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright (C) 2022 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+#include <string>
+
+#include <gtest/gtest.h>
+
+#include <ignition/common/TempDirectory.hh>
+#include <ignition/utilities/ExtraTestMacros.hh>
+
+#include "test_config.h"
+#include "test_utils.hh"
+
+#ifdef _WIN32
+  #define popen  _popen
+  #define pclose _pclose
+#endif
+
+static std::string sdf2usdCommand()
+{
+  return std::string(IGN_PATH) + "/sdf2usd";
+}
+
+/////////////////////////////////////////////////
+std::string custom_exec_str(std::string _cmd)
+{
+  _cmd += " 2>&1";
+  FILE *pipe = popen(_cmd.c_str(), "r");
+
+  if (!pipe)
+    return "ERROR";
+
+  char buffer[128];
+  std::string result = "";
+
+  while (!feof(pipe))
+  {
+    if (fgets(buffer, 128, pipe) != NULL)
+      result += buffer;
+  }
+
+  pclose(pipe);
+  return result;
+}
+
+/////////////////////////////////////////////////
+TEST(check_cmd, IGN_UTILS_TEST_DISABLED_ON_WIN32(SDF))
+{
+  std::string pathBase = PROJECT_SOURCE_PATH;
+  pathBase = ignition::common::joinPaths(pathBase, "test", "sdf");
+
+  auto tmpDir = ignition::common::tempDirectoryPath();
+  auto tmp = ignition::common::createTempDirectory("usd", tmpDir);
+  // Check a good SDF file
+  {
+    std::string path = ignition::common::joinPaths(pathBase,
+      "/shapes_world.sdf");
+    std::string output =
+      custom_exec_str(sdf2usdCommand() + " " + path + " " +
+        ignition::common::joinPaths(tmp, "shapes.usd"));
```

Comment:
When running this test locally and printing the contents of `output`, I get this:
```
sh: 1: /usr/bin/sdf2usd: not found
```

It seems like the test is checking the wrong location. Although the parser isn't completely implemented yet, I'd still imagine that the executable is generated and can be ran. Does the executable search location for this test need to be updated?

Reply:
you need to install `ignition-tools` and recompile. If you have a `ignition-tools` package already installed in your system you might need to uninstall it.

Reply:
I have the following version of `ignition-tools` installed, is it the wrong version?
```
libignition-tools-dev/unknown,now 1.4.1-1~bionic amd64 [installed,automatic]
```

I should also note that I had no issues running `UNIT_ign_TEST`, which follows a similar structure in the code.

---
 
Also, if this test requires `ignition-tools` to behave properly, do we need to add `ignition-tools` as a dependency for the tests being built? I think this line should have `ignition-tools` listed as a dependency instead of `ignition-cmake${IGN_CMAKE_VER}::utilities`:
https://github.com/ignitionrobotics/sdformat/blob/92eca6331ff67586e00e86be81e2e9d9ce125258/usd/src/CMakeLists.txt#L22

Reply:
Has this been resolved @adlarkin ?

Reply:
It should now be fixed in https://github.com/ignitionrobotics/sdformat/pull/817/commits/d91557e1095e9fd9e3cc919506a4af61e65977a5. The issue seemed to be that `IGN_PATH` is where system-wide installed binaries are (for example, the `ign` binary from `ign-tools` binary package dependency), but we need to look in the project-specific installed binaries for the component's `usd2sdf` executable. So, I replaced `IGN_PATH` with `PROJECT_BINARY_DIR`, and `usd2sdf` was found. If we shouldn't be using `PROJECT_BINARY_DIR` in this test, let me know.